content/articles/how-to-build-a-snake-game-in-javascript/index.md

```diff
@@ -0,0 +1,514 @@
+### Introduction
+
+The best way to deal with and get familiar with any programming language is through active ventures. The Snake game is a straightforward game that can be made by utilizing JavaScript and HTML abilities.
+
+Snake is a retro computer game delivered in the last part of the 1970s. The principal aim is to explore a snake while eating as many apples as could be expected while staying away from contact with the snake's body or the dividers.
+Snake is an old-school video game from the late 1970s. The primary goal is to navigate a snake while eating as many apples as possible while avoiding contact with the walls or the snake's body.
+In this instructional exercise, I'll tell you the best way to make a Snake Game utilizing JavaScript and HTML bit by bit.
+In this article, I’ll show you step-by-step how to create this Snake Game using JavaScript and HTML.
+
+### Prerequisites
+
+- Basic CSS knowledge
+- Basic javaScript knowledge
+
+### Table of contents
+
+- [Introduction](#introduction)
+- [Prerequisites](#prerequisites)
+- [Display the Canvas](#display-the-canvas)
+- [Display the snake](#display-the-snake)
+- [Use arrow keys to move and change the snake’s direction](#use-arrow-keys-to-move-and-change-the-snake’s-direction)
+- [Incorporate food](#incorporate-food)
+- [Collition detection](#collision-detection)
+- [Increase length of snake](#increase-length-of-snake)
+- [Incorporate score](#incorporate-score)
+- [Implement game over logic](#implement-game-over-logic)
+- [Add game over text](#Add-game-over-text)
+- [Conclusion](#conclusion)
+
+### Display the Canvas
+
+First, we need to display the game board and the snake. Start by creating the file `index.html`. This will contain all our code. Next, open the file in your preferred browser.
+
+To be able to create our game, we have to make use of the HTML `<canvas>`, which is used to draw graphics with JavaScript.
+
+```HTML
+ <canvas id="game" width="400" height="400"/>
+```
+
+**Note**
+The `id` is used to identify our canvas with width and height properties of 400px.
+
+Until now the browser will not display anything since the canvas has no default background. To make our canvas visible, we can give it a border by writing some CSS code.
+
+To do that, we need to insert `<style>` and `</style>` tags after the `</title>` tag. Then Copy and paste the following block of code.
+
+```CSS
+
+   <style>
+ body{
+ margin: 0px;
+ padding: 0px;
+ display: flex;
+ flex-direction: column; /* arrage items on top of the other */
+ justify-content: center;
+ align-items: center;
+       }
+ canvas{
+ box-shadow: black 20px 10px 50px; /*elevate our canvas and add shadow*/
+
+       }
+   </style>
+
+```
+
+Now, we will create an index.js file that will contain the logic. For the file to be effective on our HTML main file, we need to link the js file to the HTML file. This is done by adding the following line of code below the `</canvas>` tag.

```

Comment:
All non-English words such as file names should be distinguishable! Use, for example, backticks. Update this throughout your article!